public/components/query_compare/search_result/result_components/result_components.tsx

```diff
@@ -34,10 +35,11 @@ const InitialState = () => {
 };
 
 const ResultPanels = ({ queryResult1, queryResult2 }: ResultComponentsProps) => {
+  const { queryError1, queryError2 } = useSearchRelevanceContext();
```

Comment:
Should we extend ResultComponentsProps and retrieve query errors from ResultComponentsProps instead?

Reply:
Initially, that was how I implemented it, which is why `QueryError` was [imported in](https://github.com/opensearch-project/dashboards-search-relevance/pull/267#discussion_r1310880044).  My reasoning for changing it was that `queryError1` and `queryError2` are both already included in the `SearchRelevanceContext`. I noticed that in the rest of the codebase, if a state is included in the context, they are usually not passed in from function to function, and are used through the context instead. If I were to follow this sort of consistency, I would also need to update [result_panel.tsx](https://github.com/opensearch-project/dashboards-search-relevance/blob/de7c52dc36878b16fdc8ac8c760ffa2b97d297dc/public/components/query_compare/search_result/result_components/result_panel.tsx#L25) to grab `queryError` through context. What would be the best course of action to be consistent?

Reply:
>  I noticed that in the rest of the codebase, if a state is included in the context, they are usually not passed in from function to function, and are used through the context instead. 

I wouldn't consider it to be a pattern to follow. Is it still necessary to transmit QueryError to ResultPanel if the state of QueryError is adequately retained from SearchRelevanceContext?

Reply:
>  I noticed that in the rest of the codebase, if a state is included in the context, they are usually not passed in from function to function, and are used through the context instead. 

I believe passing props from upstream is more appropriate in this scenario.

https://github.com/opensearch-project/dashboards-search-relevance/blob/main/public/contexts/index.tsx#L17C9-L35C95 Putting everything in a global context is not always a good practice. It binds everything together, especially for data that is used in various frequency, such as the selected index, query results and query errors. There may be needless re-renders, such as when the index is changed but no query is issued.

So I would not consider this as a pattern to follow. The refactor work should be included https://github.com/opensearch-project/dashboards-search-relevance/issues/191

Reply:
Synced up offline with @noCharger. Will make changes that go towards the refactoring out of plugin: pulling things out of global context. For this PR, I'll pull `query1` and `query2` out of `SearchRelevanceContext` and pass it in from the top level (`index.tsx` -> `result_components.tsx` -> `result_panels.tsx`)